### PR TITLE
CRIMAPP-366 Add "pre order work" page

### PR DESCRIPTION
**Pull request by ripan**
## Description of change

Add "Do you intend to claim pre-order work?" page behind feature flag `means_journey`

**Add db columns**

cases. is_preorder_work_claimed(string)
cases. preorder_work_date(date)
cases. preorder_work_details(text)

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-366

## Notes for reviewer
Currently "Do you intend to claim pre-order work?" page is behind a feature flag for local and staging

### Before changes:
Screenshot 2024-01-26 at 12 55 16

### After changes:
Screenshot 2024-01-31 at 15 05 02

### CODE REVIEW

File: spec/presenters/summary/sections/case_details_spec.rb [hunk 1]
```diff
@@ -125,6 +131,116 @@
...
+        expect(answer).to be_an_instance_of(Summary::Components::ValueAnswer)
+        expect(answer.question).to eq(:is_preorder_work_claimed)
+        expect(answer.change_path).to match('applications/12345/steps/case/claim_pre_order_work')
+        expect(answer.value).to eq('no')
+
+        answer = answers[4]
+        expect(answer).to be_an_instance_of(Summary::Components::ValueAnswer)
+        expect(answer.question).to eq(:is_client_remanded)
+        expect(answer.change_path).to match('applications/12345/steps/case/has_court_remanded_client_in_custody')
+        expect(answer.value).to eq('no')
```
Comment (hiboabd) [inline spec/presenters/summary/sections/case_details_spec.rb:176]: Do we need to check all the rows here? I think we only need to test the claim pre order work row
Same comment for the spec below too

**Comment by hiboabd** [comment]
> ## Description of change
> Add "Do you intend to claim pre-order work?" page behind feature flag `means_journey`
> 
> **Add db columns**
> 
> cases. is_preorder_work_claimed(string) cases. preorder_work_date(date) cases. preorder_work_details(text)
> 
> ## Link to relevant ticket
> [dsdmoj.atlassian.net/browse/CRIMAPP-366](https://dsdmoj.atlassian.net/browse/CRIMAPP-366)
> 
> ## Notes for reviewer
> Currently "Do you intend to claim pre-order work?" page is behind a feature flag for local and staging
> 
> ## Screenshots of changes (if applicable)
> ### Before changes:
> Screenshot 2024-01-26 at 12 55 16
> ### After changes:
> Screenshot 2024-01-31 at 15 05 02
> ## How to manually test the feature

The edited screenshot was super helpful! thanks

File: spec/presenters/summary/sections/case_details_spec.rb (hunk 1 above)
Comment (ripan) [reply spec/presenters/summary/sections/case_details_spec.rb]: Make sense 👍

**Review by hiboabd — APPROVED** [review]
👍

**Pull request merged**